### ISSUE TITLE
Failed to create Karmada ins on remote cluster by using karmada operator

### ISSUE THREAD

**Issue by junfsir** [karmada-io/karmada#3783]
**What happened**:
Failed to create Karmada ins on remote cluster by using karmada operator
![image](https://github.com/karmada-io/karmada/assets/16758455/75fe5d64-8025-4fbb-acaa-efa1fcd792a9)

**Configurations**:
- Operator version: docker.io/karmada/karmada-operator:latest
- karmada object:
```
spec:
  components:
    etcd:
      local:
        imageRepository: registry.cn-hangzhou.aliyuncs.com/google_containers/etcd
        imageTag: 3.5.3-0
        replicas: 1
        resources: {}
        volumeData:
          emptyDir: {}
    karmadaAPIServer:
      imageRepository: registry.cn-hangzhou.aliyuncs.com/google_containers/kube-apiserver
      imageTag: v1.25.4
      replicas: 1
      resources: {}
      serviceSubnet: 10.96.0.0/12
      serviceType: ClusterIP
    karmadaAggregatedAPIServer:
      imageRepository: docker.io/karmada/karmada-aggregated-apiserver
      imageTag: v1.6.0
      replicas: 1
      resources: {}
    karmadaControllerManager:
      imageRepository: docker.io/karmada/karmada-controller-manager
      imageTag: v1.6.0
      replicas: 1
      resources: {}
    karmadaScheduler:
      imageRepository: docker.io/karmada/karmada-scheduler
      imageTag: v1.6.0
      replicas: 1
      resources: {}
    karmadaWebhook:
      imageRepository: docker.io/karmada/karmada-webhook
      imageTag: v1.6.0
      replicas: 1
      resources: {}
    kubeControllerManager:
      imageRepository: registry.cn-hangzhou.aliyuncs.com/google_containers/kube-controller-manager
      imageTag: v1.25.4
      replicas: 1
      resources: {}
  hostCluster:
    apiEndpoint: 10.2.10.169:6443
    networking:
      dnsDomain: cluster.local
    secretRef:
      name: poc2
      namespace: test
```
- Secret
![image](https://github.com/karmada-io/karmada/assets/16758455/9c6d34b3-5a2f-4108-9f1c-6c3f6d61cb64)

**Comment by RainbowMango** [comment]
Thanks @junfsir for spotting.
cc operator owner here @calvin0327 @Poor12 for help

**Comment by calvin0327** [comment]
@junfsir Yes, now, we seem not support install karmada on remote cluster.

Thanks for pointing this out, I wonder if you are using karmada operator and have this scenario to install karmada on a remote cluster?

**Comment by junfsir** [comment]
@calvin0327 It would be better to support it. Does the community have plans to support this feature？

**Comment by calvin0327** [comment]
> @calvin0327 It would be better to support it. Does the community have plans to support this feature？

That's sure. 
/assign

**Comment by RainbowMango** [comment]
I'm trying to figure out which PR/Issue should be included in the coming v1.7 release which is planned at the end of this month.

@calvin0327 are you still working on #3926?  that has some lint issues that need to be resolved.
